libs/api/domains/powerbi/src/lib/powerbi.resolver.ts

```diff
@@ -0,0 +1,176 @@
+import { Configuration, ConfidentialClientApplication } from '@azure/msal-node'
+import { Inject } from '@nestjs/common'
+import { Args, Query, Resolver } from '@nestjs/graphql'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import { PowerBiEmbedTokenInput } from './dto/powerbiEmbedToken.input'
+import { PowerBiEmbedTokenResponse } from './models/powerbiEmbedTokenResponse'
+
+type Owner = 'Fiskistofa'
+const BASE_URL = 'https://api.powerbi.com/v1.0/myorg'
+
+@Resolver()
+export class PowerBiResolver {
+  constructor(
+    @Inject(LOGGER_PROVIDER)
+    private readonly logger: Logger,
+  ) {}
+
+  private async getAccessToken(owner: Owner) {
```

Comment:
Should keep the resolver class light and move the logic to a service class.

Reply:
I agree, hopefully I was able to move things over to a service class in a "correct manner"